File: Java/Stack/BJAlgo_10828.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class BJAlgo_10828 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int count = Integer.parseInt(br.readLine());
        List stack = new ArrayList();

        for(int i = 0; i < count; i++) {
            String[] comm = br.readLine().split(" ");

            switch (comm[0]) {
                case "push":
                    stack.add(Integer.parseInt(comm[1]));
                    break;
                case "pop":
                    if (stack.size() > 0) {
                        int data = (int) stack.get(stack.size() - 1);
                        stack.remove(stack.size() - 1);
                        System.out.println(data);
                    } else {
                        System.out.println(-1);
                    }
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    if (stack.isEmpty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;
                case "top":
                    if (stack.size() > 0) {
                        System.out.println(stack.get(stack.size() - 1));
                    } else {
                        System.out.println(-1);
                    }
                    break;
            }
        }
    }
}

File: Python/Stack/BJAlgo_10828.py
from sys import stdin

count = int(stdin.readline())
stack = []

for _ in range(count):
    comm = stdin.readline().split()

    if comm[0] == 'push':
        stack.append(int(comm[1]))
    elif comm[0] == 'pop':
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif comm[0] == 'size':
        print(len(stack))
    elif comm[0] == 'empty':
        print(1 - int(bool(stack)))
    elif comm[0] == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)